File: ella/week1/H_2559/2559.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] arr = new int[n];

        arr[0] = sc.nextInt();
        for(int i=1; i<n; i++){
          arr[i] = arr[i-1] + sc.nextInt();
        }

        int max = arr[k-1];
        for(int i=k; i<n; i++){
            int sum = arr[i] - arr[i-k];
            if(sum > max) max = sum;
        }

        System.out.print(max);
    }
}

File: dorothy/week1/2559.py
'''
1: N K
    N: 온도를 측정한 전체 날짜의 수 (2 <= N <= 100000)
    K: 연속적인 날짜의 수 (1<k<N)
2: N개의 정수 빈 칸을 사이에 두고 주어짐
'''
N, K = [int(x) for x in input().split()]
t = [int(x) for x in input().split()]

s = [sum(t[:K])]
for i in range(1, N-K+1):
    s.append(s[i-1]- t[i-1] + t[i+K-1])

print(max(s))
